File: 99_StrSymmetryPoint.cpp
# include <algorithm>

int solution(string &S)
{
	if (S.length()%2==0)
		return -1;
	else if (S.length()==1)
		return 0;
	else
	{
		string s2=S;
		reverse(s2.begin(),s2.end());
		if (s2==S)
			return s2.length()/2;
		else
			return -1;
	}
}

File: 99_StrSymmetryPoint.py
def solution(S):
	if (len(S)%2==0):
		return -1
	elif (len(S)==1):
		return 0
	else:
		S2=S[-1:-len(S)-1:-1]
		if (S == S2):
			return len(S)//2 
		else:
			return -1

